<ol>
    <li><a href="#clone-01">CLONE - 01</a></li>
    <li><a href="#clone-02">CLONE - 02</a></li>
    <li><a href="#push-01">PUSH - 01</a></li>
    <li><a href="#push-02">PUSH - 02</a></li>
    <li><a href="#push-03">PUSH - 03</a></li>
    <li><a href="#push-04">PUSH - 04</a></li>
    <li><a href="#push-05">PUSH - 05</a></li>
    <li><a href="#pull-01">PULL - 01</a></li>
    <li><a href="#pull-02">PULL - 02</a></li>
    <li><a href="#repositorio-01">REPOSITOROIO - 01</a></li>
    <li><a href="#"></a></li>
</ol>

<!-- ------------------------------------------------------------------- -->
<hr>
<a name="clone-01"></a>
<h2>CLONE - 01</h2>

<br/>erro de nao achar o arquivo acontece quando esta clonando o projeto<br/>

<code>Unlink of file 'templates/media/container.html' failed. Should I try again? (y/n)</code>
<br/>

Isso pode significar que outro programa está usando o arquivo, o que está impedindo o git de "mover" o arquivo para dentro ou para fora do diretório de trabalho quando você está tentando alterar as ramificações.
<br/><br/>

Nesse caso, tente fechar o arquivo em qualquer aplicativo que possa tê-lo usado. Se isso não funcionar, saia completamente de qualquer aplicativo que possa ter aberto o arquivo.
<br/><br/>

Parei o servidor<code>$ ionic serve</code>, fechei o terminal do mesmo esperei alguns minutos ,execeutei o <code>git clone</code>
e funcionou \o/

<br><a href="https://stackoverflow.com/questions/4389833/unlink-of-file-failed-should-i-try-again">stackoverflow</a>

<!-- ------------------------------------------------------------------- -->
<hr>
<a name="clone-02"></a>
<h2>CLONE - 02</h2>
<code>fatal: update_ref failed for ref 'HEAD': cannot lock ref 'HEAD': unable to resolve reference</code>
<br><br>
SOLUCAO:
<ol>
    <li>criar o diretorio manualmente
        <pre class="terminal">$ sudo mkdir PROJETO_NOME</pre>
    </li>
    <li>startar git
        <pre class="terminal">$ git ini</pre>
    </li>
    <li>setar repositorio
        <pre class="terminal">$ git remote add origin PROJETO_REPOSITORIO</pre>
    </li>
    <li>buscar arquivos da branch padrao(master)
        <pre class="terminal">$ git pull</pre>
    </li>
    <li>largar esses arquivos na memoria local
        <pre class="terminal">$ git add .</pre>
        <pre class="terminal">$ git commit -m "temp"</pre>
    </li>    
    <li>listar branchs remotas
        <pre class="terminal">$ git branch -a</pre>
    </li>  
    <li>buscar branch
        <pre class="terminal">$ git checkout -t remote/origin/PROJETO_BRANCH</pre>
    </li>  
</ol>

CONCLUIDO !

<!-- ------------------------------------------------------------------- -->
<hr>
<a name="push-01"></a>
<h2>PUSH - 01</h2>
<code>Unable to rewind rpc post data - try increasing http.postBuffer error: <br/>RPC failed; result=65, HTTP code = 301</code>
<BR/>Solução encontrada:<br>
<pre class="terminal">$ git config http.postBuffer 524288000</pre>
<a href="https://stackoverflow.com/questions/2702731/git-fails-when-pushing-commit-to-github">stackoverflow</a>

<!-- ------------------------------------------------------------------- -->
<hr>
<a name="push-02"></a>
<h2>PUSH - 02</h2>
<code>fatal: unable to access 'https://github.com/scarapa/ajuda.git/': Problem with the SSL CA cert (path? access rights?)</code>
<br><br>
Solucao
<pre class="terminal">
$ git config --global http.sslVerify false
$ git config http.sslVerify false    
</pre>

<!-- ------------------------------------------------------------------- -->
<hr>
<a name="push-03"></a>
<h2>PUSH - 03</h2>
Mensagem de erro:<br/>
<code>To prevent you from losing history, non-fast-forward updates were rejected Merge the remote changes (e.g. 'git pull') before pushing again.  See the 'Note about fast-forwards' section of 'git push --help' for details.</code>

<br/>Isso aconteceu depois que tentei fazer o <b>COMMIT</b> antes do <b>PULL</b>
Solução:<br/>
<pre class="terminal">$ git push --force</pre>

<a href="https://stackoverflow.com/questions/3598355/i-am-not-able-to-push-on-git">stackoverflow</a> - 
<a href="http://blog.locaweb.com.br/artigos/git-alem-do-push-e-pull-dicas-pra-voce-que-chegou-na-semana-passada/">Fonte 2</a>

<!-- ------------------------------------------------------------------- -->
<hr>
<a name="push-04"></a>
<h2>PUSH - 04</h2>
<code>fatal: refusing to merge unrelated histories</code>
<br><br>

A solução para conseguir fazer o push foi trazer o projeto com o comando abaixo
<pre class="terminal">$ git pull origin master --allow-unrelated-histories</pre>
<a href="https://github.community/t5/How-to-use-Git-and-GitHub/How-to-deal-with-quot-refusing-to-merge-unrelated-histories-quot/td-p/12619" target="_blank">Fonte</a>

<!-- ------------------------------------------------------------------- -->
<hr>
<a name="push-05"></a>
<h2>PUSH - 05</h2>
<code>error: src refspec master does not match any</code>
<br><br>
Problema é que os nomes do repositorio <code>LOCAL</code> e <code>REMOTO</code> não combinam

<a href="https://pt.stackoverflow.com/questions/21243/problema-pra-publicar-projeto-github">stackoverflow</a>



<!-- ------------------------------------------------------------------- -->
<hr>
<a name="push-06"></a>
<h2>PUSH - 06 - TAMAHO EXCEDIDO</h2>

<pre class="terminal" style="font-weight: bold;">$ git push
Delta compression using up to 12 threads
Compressing objects: 100% (90/90), done.
Writing objects: 100% (93/93), 312.75 MiB | 5.66 MiB/s, done.
Total 93 (delta 59), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (59/59), completed with 13 local objects.
remote: <span style="color:rgb(239, 103, 103)">error</span>: Trace: 199b2c1df1f57b2ae009620697b8e9cef84dd5ffc10135365958e6db348111a9
remote: <span style="color:rgb(239, 103, 103)">error</span>: File arquivos/tools_r25.2.3-windows.zip is 292.54 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: <span style="color:rgb(239, 103, 103)">error</span>: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/scarapa/html_procergs.git
<span style="color:rgb(197, 0, 0)">! [remote rejected]</span> desenv -> desenv (pre-receive hook declined)
<span style="color:rgb(197, 0, 0)">error: failed to push some refs to 'https://github.com/scarapa/html_procergs.git'</span></pre>

<pre class="terminal">$ git filter-branch</pre>

<pre class="terminal">$ git merge</pre>

<br><a href="https://docs.github.com/pt/repositories/working-with-files/managing-large-files/about-large-files-on-github">docs.github</a>

<!-- ------------------------------------------------------------------- -->
<hr>
<a name="pull-01"></a>
<h2>PULL - 01</h2>
Erro completo abaixo:
<pre>error: Your local changes to the following files would be overwritten by merge
*** LISTA DE ARQUIVOS ***
Please commit your changes or stash them before you merge.
Aborting</pre>

Acontece porque a sua versão é anterior a verão atual, portanto antes de fazer o commit e o push você precisa atualizar para que os conflitos sejam locais
e nao no server.<br><br>

Solução:<br/>
<pre class="terminal">$ git checkout HEAD^ CAMINHO/DO/ARQUIVO
$ git pull</pre>
<a href="https://stackoverflow.com/questions/14318234/how-to-ignore-error-on-git-pull-about-my-local-changes-would-be-overwritten-by-m">stackoverflow</a>


<!-- ------------------------------------------------------------------- -->
<hr>
<a name="pull-02"></a>
<h2>PULL - 02</h2>
Mensagem de erro:<br/>
<code>git error  cannot lock ref 'refs/remotes/origin/master':</code>
<pre class="terminal">$ git gc --prune=now
$ git remote prune origin
$ git pull origin master</pre>
<a href="https://stackoverflow.com/questions/2998832/git-pull-fails-unable-to-resolve-reference-unable-to-update-local-ref">Fonte</a>

<!-- ------------------------------------------------------------------- -->
<hr>
<a name="repositorio-01"></a>
<h2>REPOSITORIO - 01</h2>
<code>Repository NOT FOUND !!!</code>
<br>
<h3>verificar sem tem permissão !!!</h3>
<pre class="terminal">$ ssh -T git@github.com</pre>
<code>retorno</code>
<pre>> Hi username! You've successfully authenticated, but GitHub does not
> provide shell access.</pre>

<h3>caso tenha, REMOVER repositorio atual</h3>
<pre class="terminal">$ git remote rm origin</pre>

<h3>ADICIONAR repositorio</h3>
<pre class="terminal">$ git remote add origin  https://<b style="color:red">USERNAME:PASSWORD</b>@github.com/username/reponame.git</pre>

<a href="https://help.github.com/en/github/creating-cloning-and-archiving-repositories/error-repository-not-found">help.github</a><br>
<a href="https://stackoverflow.com/questions/10116373/git-push-error-repository-not-found">stackoverflow</a><br>




